### PR TITLE
Docs: Add a color mode switcher + a sticky header on all pages

### PR DESCRIPTION
**Pull request by louismaximepiton**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Related issues

NA

### Description

Added:
- Color mode switcher javascript from Boosted
- Stick the header to the top
- Add the color mode switcher in the header
- Tweak a bit the CSS

### Motivation & Context

Easier to test dark mode handling in the documentation using this way.

### Types of change

- New feature (non-breaking change which adds functionality)

### Test checklist

Please check that the following tests projects are still working:

- [x] `docs/examples`
- [x] `test/angular-ngx-echarts`
- [x] `test/angular-echarts`
- [x] `test/html`
- [x] `test/react`
- [x] `test/vue`
- [x] `test/examples/bar-line-chart`
- [x] `test/examples/single-line-chart`
- [x] `test/examples/timeseries-chart`

### CODE REVIEW

**Review by jacques-lebourgeois — APPROVED** [review]
if it suits you, I added a commit to keep [use case example in light mode](https://github.com/Orange-OpenSource/ods-charts/pull/421/commits/3c41c75a5db6a45f5c8982f0d4052ae344618f3e).

The rest is fine with me too.

**Comment by louismaximepiton** [comment]
Looks good, thanks!

File: docs/examples/index.js [hunk 1]
```diff
@@ -667,6 +667,9 @@ window.downloadTheme = downloadTheme;
 window.generateSingleLineChart = async (id) => {
   // Specify the configuration items and data for the chart
   var option = {
+    title: {
+      text: 'test',
+    },
```
Comment (jacques-lebourgeois) [inline docs/examples/index.js:672]: I assume it's to be removed?

File: docs/assets/style.css [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  min-height: 100vh;
+}
+
+.footer {
+  position: sticky;
+  top: 100vh;
+}
+
+.border-subtle {
+  --bs-border-color: var(--bs-border-color-subtle);
```
Comment (jacques-lebourgeois) [inline docs/assets/style.css:12]: Why did you need this class in place of border-light like describe on https://boosted.orange.com/docs/5.3/utilities/borders/ ?

File: docs/assets/style.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+html,
+body {
+  min-height: 100vh;
+}
```
Comment (jacques-lebourgeois) [inline docs/assets/style.css:4]: could we just use min-vh-100 in the HTML ?

File: docs/assets/style.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+.footer {
+  position: sticky;
+  top: 100vh;
+}
```
Comment (jacques-lebourgeois) [inline docs/assets/style.css:9]: could we just use class="min-vh-100 position-sticky" in the HTML ?

**Review by jacques-lebourgeois — CHANGES_REQUESTED** [review]
I notice that when switching of mode, it scroll tos the top!
scroll_to_top

File: docs/assets/style.css (hunk 2 above)
Comment (jacques-lebourgeois) [reply docs/assets/style.css:12]: You give me the explanation that you wanted to border color to be adapt to the dark mode

File: docs/assets/style.css (hunk 3 above)
Comment (jacques-lebourgeois) [reply docs/assets/style.css:4]: This is an optional comment. As there is a style.css file anyway, no problem to keep thes rules

File: docs/assets/style.css (hunk 4 above)
Comment (jacques-lebourgeois) [reply docs/assets/style.css:9]: This is an optional comment. As there is a style.css file anyway, no problem to keep thes rules

**Comment by louismaximepiton** [comment]
> I notice that when switching of mode, it scrolls to the top! scroll_to_top [ scroll_to_top ](https://private-user-images.githubusercontent.com/40755890/395194361-a6f85ca0-068c-47da-addc-5d5bb3c5e039.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwODE0NTgsIm5iZiI6MTczNDA4MTE1OCwicGF0aCI6Ii80MDc1NTg5MC8zOTUxOTQzNjEtYTZmODVjYTAtMDY4Yy00N2RhLWFkZGMtNWQ1YmIzYzVlMDM5LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEzVDA5MTIzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyZWNmNzBmY2NlMjE5ZDVkOWFjODY1MzljMDNiY2E2OWJjZjVkZjQ5NGVjODExNzk3ODE4ODM1ZTMyYzQwZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ACSHLKk6CPeTaHZ1_qMOhElJV8xalyzit--b-2e6OKE) [ ](https://private-user-images.githubusercontent.com/40755890/395194361-a6f85ca0-068c-47da-addc-5d5bb3c5e039.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwODE0NTgsIm5iZiI6MTczNDA4MTE1OCwicGF0aCI6Ii80MDc1NTg5MC8zOTUxOTQzNjEtYTZmODVjYTAtMDY4Yy00N2RhLWFkZGMtNWQ1YmIzYzVlMDM5LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEzVDA5MTIzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyZWNmNzBmY2NlMjE5ZDVkOWFjODY1MzljMDNiY2E2OWJjZjVkZjQ5NGVjODExNzk3ODE4ODM1ZTMyYzQwZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ACSHLKk6CPeTaHZ1_qMOhElJV8xalyzit--b-2e6OKE)

Yeah it's related to https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues/1240. If you're fine we can merge as-is and postpone the factorization of the documentation ?

File: docs/examples/index.js (hunk 1 above)
Comment (louismaximepiton) [reply docs/examples/index.js]: Good catch!

**Pull request merged**